core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks.go

```diff
@@ -0,0 +1,828 @@
+    if err != nil {
+        log.Error(err.Error())
+        return false, err
+    }
+    iccResp := stub.InvokeChaincode(am.interopChaincodeId, [][]byte{[]byte("LockAsset"), assetAgreementBytes, lockInfoBytes}, "")
+    fmt.Printf("Response from Interop CC: %+v\n", iccResp)
+    if iccResp.GetStatus() != shim.OK {
+        return false, fmt.Errorf(string(iccResp.GetPayload()))
+    }
+    lockExpiryTimeSec := int64(lockInfoHTLC.ExpiryTimeMillis)/1000
+    lockExpiryTimeNano := (int64(lockInfoHTLC.ExpiryTimeMillis) - (lockExpiryTimeSec * 1000)) * (1000 * 1000)     // Convert milliseconds to nanoseconds
+    fmt.Printf("Asset %s of type %s locked for %s until %+v\n", assetAgreement.Id, assetAgreement.Type, assetAgreement.Recipient, time.Unix(lockExpiryTimeSec, lockExpiryTimeNano))
+    return true, nil
+}
+
+func (am *AssetManagement) LockFungibleAsset(stub shim.ChaincodeStubInterface, assetAgreement *common.FungibleAssetAgreement, lockInfo *common.AssetLock) (bool, error) {
+    var errorMsg string
+
+    if len(am.interopChaincodeId) == 0 {
+        errorMsg = "Interoperation chaincode ID not set. Run the 'Configure(...) function first."
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+
+    if (lockInfo.LockMechanism != common.LockMechanism_HTLC) {
+        log.Errorf("Unsupported lock mechanism: %+v", lockInfo.LockMechanism)
+        return false, fmt.Errorf("Unsupported lock mechanism: %+v", lockInfo.LockMechanism)
+    }
+    lockInfoHTLC := &common.AssetLockHTLC{}
+    if len(lockInfo.LockInfo) == 0 {
+        log.Error("Empty lock info")
+        return false, fmt.Errorf("Empty lock info")
+    }
+    err := proto.Unmarshal(lockInfo.LockInfo, lockInfoHTLC)
+    if err != nil {
+        log.Error(err.Error())
+        return false, err
+    }
+    if lockInfoHTLC.TimeSpec != common.AssetLockHTLC_EPOCH {
+        log.Error("Only EPOCH time is supported at present")
+        return false, fmt.Errorf("Only EPOCH time is supported at present")
+    }
+
+    if len(assetAgreement.Type) == 0 {
+        errorMsg = "Empty asset type"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if assetAgreement.NumUnits <= 0 {
+        errorMsg = "Invalid number of asset units"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if len(assetAgreement.Recipient) == 0 {
+        errorMsg = "Empty lock recipient"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    if len(lockInfoHTLC.Hash) == 0 {
+        errorMsg = "Empty lock hash value"
+        log.Error(errorMsg)
+        return false, errors.New(errorMsg)
+    }
+    assetAgreementBytes, err := proto.Marshal(assetAgreement)
+    if err != nil {
+        log.Error(err.Error())
+        return false, err
+    }
+    lockInfoBytes, err := proto.Marshal(lockInfoHTLC)
+    if err != nil {
+        log.Error(err.Error())
+        return false, err
+    }
```

Comment:
There is a significant amount of redundant boilerplate in how we perform validations of args in each function, which reduces readability and probably maintainability. I suggest we refactor to consolidate common validation logic in separate reusable helper functions and minimize validation boilerplate in the core API methods.

Reply:
Whether or not argument validation separation from the main function makes the code more readable is up for debate in my opinion.

I do agree that boilerplate code redundancy is not a great thing, but different functions perform validations on different parameter combinations, so I don't believe we will end up creating a small set of helper functions. Rather, each helper function will likely check for "emptiness" of a single parameter.

But I'll review the code once again to determine if such refactoring is useful.